### ISSUE TITLE
Flash loader broken in V1_feature_branch

### ISSUE THREAD

**Issue by rettigl** [OpenCOMPES/sed#492]
When converting the 62 files from the flash loader example in the v1_feature_branch, I get the following error:
```
System config loaded from: [/mnt/pcshare/users/Laurenz/AreaB/sed/sed/sed/config/flash_example_config.yaml]
Default config loaded from: [/mnt/pcshare/users/Laurenz/AreaB/sed/sed/sed/config/default.yaml]
INFO - Reading files: 62 new files of 62 total.
[Parallel(n_jobs=62)]: Using backend LokyBackend with 62 concurrent workers.
[Parallel(n_jobs=62)]: Done   2 out of  62 | elapsed:   27.3s remaining: 13.7min
---------------------------------------------------------------------------
_RemoteTraceback                          Traceback (most recent call last)
_RemoteTraceback: 
"""
Traceback (most recent call last):
  File "/mnt/pcshare/users/Laurenz/AreaB/sed/poetry_envs/virtualenvs/sed-processor-3qnpZCFI-py3.9/lib/python3.9/site-packages/joblib/externals/loky/process_executor.py", line 463, in _process_worker
    r = call_item()
  File "/mnt/pcshare/users/Laurenz/AreaB/sed/poetry_envs/virtualenvs/sed-processor-3qnpZCFI-py3.9/lib/python3.9/site-packages/joblib/externals/loky/process_executor.py", line 291, in __call__
    return self.fn(*self.args, **self.kwargs)
  File "/mnt/pcshare/users/Laurenz/AreaB/sed/poetry_envs/virtualenvs/sed-processor-3qnpZCFI-py3.9/lib/python3.9/site-packages/joblib/parallel.py", line 598, in __call__
    return [func(*args, **kwargs)
  File "/mnt/pcshare/users/Laurenz/AreaB/sed/poetry_envs/virtualenvs/sed-processor-3qnpZCFI-py3.9/lib/python3.9/site-packages/joblib/parallel.py", line 598, in <listcomp>
    return [func(*args, **kwargs)
  File "/mnt/pcshare/users/Laurenz/AreaB/sed/sed/sed/loader/flash/buffer_handler.py", line 151, in _save_buffer_file
    df.dropna(subset=electron_channels).astype(dtypes).reset_index().to_parquet(
  File "/mnt/pcshare/users/Laurenz/AreaB/sed/poetry_envs/virtualenvs/sed-processor-3qnpZCFI-py3.9/lib/python3.9/site-packages/pandas/core/generic.py", line 6620, in astype
    res_col = col.astype(dtype=cdt, copy=copy, errors=errors)
  File "/mnt/pcshare/users/Laurenz/AreaB/sed/poetry_envs/virtualenvs/sed-processor-3qnpZCFI-py3.9/lib/python3.9/site-packages/pandas/core/generic.py", line 6643, in astype
    new_data = self._mgr.astype(dtype=dtype, copy=copy, errors=errors)
  File "/mnt/pcshare/users/Laurenz/AreaB/sed/poetry_envs/virtualenvs/sed-processor-3qnpZCFI-py3.9/lib/python3.9/site-packages/pandas/core/internals/managers.py", line 430, in astype
    return self.apply(
  File "/mnt/pcshare/users/Laurenz/AreaB/sed/poetry_envs/virtualenvs/sed-processor-3qnpZCFI-py3.9/lib/python3.9/site-packages/pandas/core/internals/managers.py", line 363, in apply
    applied = getattr(b, f)(**kwargs)
  File "/mnt/pcshare/users/Laurenz/AreaB/sed/poetry_envs/virtualenvs/sed-processor-3qnpZCFI-py3.9/lib/python3.9/site-packages/pandas/core/internals/blocks.py", line 758, in astype
    new_values = astype_array_safe(values, dtype, copy=copy, errors=errors)
  File "/mnt/pcshare/users/Laurenz/AreaB/sed/poetry_envs/virtualenvs/sed-processor-3qnpZCFI-py3.9/lib/python3.9/site-packages/pandas/core/dtypes/astype.py", line 237, in astype_array_safe
    new_values = astype_array(values, dtype, copy=copy)
  File "/mnt/pcshare/users/Laurenz/AreaB/sed/poetry_envs/virtualenvs/sed-processor-3qnpZCFI-py3.9/lib/python3.9/site-packages/pandas/core/dtypes/astype.py", line 182, in astype_array
    values = _astype_nansafe(values, dtype, copy=copy)
  File "/mnt/pcshare/users/Laurenz/AreaB/sed/poetry_envs/virtualenvs/sed-processor-3qnpZCFI-py3.9/lib/python3.9/site-packages/pandas/core/dtypes/astype.py", line 101, in _astype_nansafe
    return _astype_float_to_int_nansafe(arr, dtype, copy)
  File "/mnt/pcshare/users/Laurenz/AreaB/sed/poetry_envs/virtualenvs/sed-processor-3qnpZCFI-py3.9/lib/python3.9/site-packages/pandas/core/dtypes/astype.py", line 151, in _astype_float_to_int_nansafe
    raise ValueError(f"Cannot losslessly cast from {values.dtype} to {dtype}")
ValueError: Cannot losslessly cast from float32 to uint16: Error while type casting for column 'dldPosX'
"""

The above exception was the direct cause of the following exception:

ValueError                                Traceback (most recent call last)
Cell In[19], line 1
----> 1 sp = SedProcessor(
      2     runs=[44824,44825,44826,44827],
      3     config=config_override,
      4     system_config=config_file,
      5     collect_metadata=False,
      6     force_recreate=True,
      7 )

File /mnt/pcshare/users/Laurenz/AreaB/sed/sed/sed/core/processor.py:166, in SedProcessor.__init__(self, metadata, config, dataframe, files, folder, runs, collect_metadata, verbose, **kwds)
    164 # Load data if provided:
    165 if dataframe is not None or files is not None or folder is not None or runs is not None:
--> 166     self.load(
    167         dataframe=dataframe,
    168         metadata=metadata,
    169         files=files,
    170         folder=folder,
    171         runs=runs,
    172         collect_metadata=collect_metadata,
    173         **kwds,
    174     )

File /mnt/pcshare/users/Laurenz/AreaB/sed/sed/sed/core/processor.py:416, in SedProcessor.load(self, dataframe, metadata, files, folder, runs, collect_metadata, **kwds)
    408         dataframe, timed_dataframe, metadata = self.loader.read_dataframe(
    409             folders=cast(str, self.cpy(folder)),
    410             runs=runs,
   (...)
    413             **kwds,
    414         )
    415     else:
--> 416         dataframe, timed_dataframe, metadata = self.loader.read_dataframe(
    417             runs=runs,
    418             metadata=metadata,
    419             collect_metadata=collect_metadata,
    420             **kwds,
    421         )
    423 elif folder is not None:
    424     dataframe, timed_dataframe, metadata = self.loader.read_dataframe(
    425         folders=cast(str, self.cpy(folder)),
    426         metadata=metadata,
    427         collect_metadata=collect_metadata,
    428         **kwds,
    429     )

File /mnt/pcshare/users/Laurenz/AreaB/sed/sed/sed/loader/flash/loader.py:338, in FlashLoader.read_dataframe(self, files, folders, runs, ftype, metadata, collect_metadata, detector, force_recreate, processed_dir, debug, **kwds)
    335 # Obtain the parquet filenames, metadata, and schema from the method
    336 # which handles buffer file creation/reading
    337 h5_paths = [Path(file) for file in self.files]
--> 338 df, df_timed = bh.process_and_load_dataframe(
    339     h5_paths=h5_paths,
    340     folder=processed_dir,
    341     force_recreate=force_recreate,
    342     suffix=detector,
    343     debug=debug,
    344 )
    346 if self.instrument == "wespe":
    347     df, df_timed = wespe_convert(df, df_timed)

File /mnt/pcshare/users/Laurenz/AreaB/sed/sed/sed/loader/flash/buffer_handler.py:308, in BufferHandler.process_and_load_dataframe(self, h5_paths, folder, force_recreate, suffix, debug, remove_invalid_files)
    298     schema_set = set(
    299         get_channels(
    300             self._config,
   (...)
    304         ),
    305     ) - {"electronId"}
    306     self._schema_check(self.fp["timed"], schema_set)
--> 308 self._save_buffer_files(force_recreate, debug)
    310 self._get_dataframes()
    312 return self.df["electron"], self.df["timed"]

File /mnt/pcshare/users/Laurenz/AreaB/sed/sed/sed/loader/flash/buffer_handler.py:211, in BufferHandler._save_buffer_files(self, force_recreate, debug)
    209         self._save_buffer_file(file_set)
    210 else:
--> 211     Parallel(n_jobs=n_cores, verbose=10)(
    212         delayed(self._save_buffer_file)(file_set) for file_set in file_sets
    213     )

File /mnt/pcshare/users/Laurenz/AreaB/sed/poetry_envs/virtualenvs/sed-processor-3qnpZCFI-py3.9/lib/python3.9/site-packages/joblib/parallel.py:2007, in Parallel.__call__(self, iterable)
   2001 # The first item from the output is blank, but it makes the interpreter
   2002 # progress until it enters the Try/Except block of the generator and
   2003 # reaches the first `yield` statement. This starts the asynchronous
   2004 # dispatch of the tasks to the workers.
   2005 next(output)
-> 2007 return output if self.return_generator else list(output)

File /mnt/pcshare/users/Laurenz/AreaB/sed/poetry_envs/virtualenvs/sed-processor-3qnpZCFI-py3.9/lib/python3.9/site-packages/joblib/parallel.py:1650, in Parallel._get_outputs(self, iterator, pre_dispatch)
   1647     yield
   1649     with self._backend.retrieval_context():
-> 1650         yield from self._retrieve()
   1652 except GeneratorExit:
   1653     # The generator has been garbage collected before being fully
   1654     # consumed. This aborts the remaining tasks if possible and warn
   1655     # the user if necessary.
   1656     self._exception = True

File /mnt/pcshare/users/Laurenz/AreaB/sed/poetry_envs/virtualenvs/sed-processor-3qnpZCFI-py3.9/lib/python3.9/site-packages/joblib/parallel.py:1754, in Parallel._retrieve(self)
   1747 while self._wait_retrieval():
   1748 
   1749     # If the callback thread of a worker has signaled that its task
   1750     # triggered an exception, or if the retrieval loop has raised an
   1751     # exception (e.g. `GeneratorExit`), exit the loop and surface the
   1752     # worker traceback.
   1753     if self._aborting:
-> 1754         self._raise_error_fast()
   1755         break
   1757     # If the next job is not ready for retrieval yet, we just wait for
   1758     # async callbacks to progress.

File /mnt/pcshare/users/Laurenz/AreaB/sed/poetry_envs/virtualenvs/sed-processor-3qnpZCFI-py3.9/lib/python3.9/site-packages/joblib/parallel.py:1789, in Parallel._raise_error_fast(self)
   1785 # If this error job exists, immediately raise the error by
   1786 # calling get_result. This job might not exists if abort has been
   1787 # called directly or if the generator is gc'ed.
   1788 if error_job is not None:
-> 1789     error_job.get_result(self.timeout)

File /mnt/pcshare/users/Laurenz/AreaB/sed/poetry_envs/virtualenvs/sed-processor-3qnpZCFI-py3.9/lib/python3.9/site-packages/joblib/parallel.py:745, in BatchCompletionCallBack.get_result(self, timeout)
    739 backend = self.parallel._backend
    741 if backend.supports_retrieve_callback:
    742     # We assume that the result has already been retrieved by the
    743     # callback thread, and is stored internally. It's just waiting to
    744     # be returned.
--> 745     return self._return_or_raise()
    747 # For other backends, the main thread needs to run the retrieval step.
    748 try:

File /mnt/pcshare/users/Laurenz/AreaB/sed/poetry_envs/virtualenvs/sed-processor-3qnpZCFI-py3.9/lib/python3.9/site-packages/joblib/parallel.py:763, in BatchCompletionCallBack._return_or_raise(self)
    761 try:
    762     if self.status == TASK_ERROR:
--> 763         raise self._result
    764     return self._result
    765 finally:

ValueError: Cannot losslessly cast from float32 to uint16: Error while type casting for column 'dldPosX'
```

This also occurs in the v1 branch, and is probably related to the recently introduced dtypes, which might not have been tested properly.
When I remove dtype entries from the config, it works.

**Comment by zain-sohail** [comment]
Yes, I noticed this issue before too. Seems to occur with only some h5 files. Likely there is a float value in the column, which should be removed anyways. I will take a look what is causing that problem.

**Comment by zain-sohail** [comment]
<img width="480" alt="image" src="https://github.com/user-attachments/assets/3d06d4d1-4093-449f-af68-060575be4a07">

This seems to be the reason.
I have added a filter in dataframe class to remove all pulse values below 0 as they are anyways invalid.

**Comment by zain-sohail** [comment]
closed by #484
